content/blog/2025/07/21/2025-07-21-redesigning-jenkins-part-two.adoc

```diff
@@ -0,0 +1,358 @@
+        background-image: var(--aurora);
+        opacity: 0.35;
+        z-index: -1;
+      }
+
+      &::after {
+        content: "";
+        position: absolute;
+        inset: 0;
+        border: 1.5px solid color-mix(in srgb, var(--color) 5%, transparent);
+        border-radius: 10px;
+        z-index: 10;
+      }
+    }
+
+    .developer-exp {
+      display: flex;
+      align-items: center;
+      justify-content: center;
+      gap: 40px;
+
+      img {
+        width: 90px;
+        height: 90px;
+
+        &:nth-of-type(3) {
+          scale: 0.9;
+        }
+
+        &:nth-of-type(even) {
+          width: 40px;
+          height: 40px;
+          rotate: 45deg;
+          mix-blend-mode: overlay;
+        }
+      }
+    }
+</style>
+
+<div class="app-blog-image app-blog-image--masthead">
+    <div class="app-jenkins-preview"></div>
+    <div class="app-jenkins-preview"></div>
+    <div class="app-jenkins-preview"></div>
+    <div class="app-jenkins-preview"></div>
+    <div class="app-jenkins-preview"></div>
+</div>
+
+<script>
+(function () {
+  const startDeg = [-12, -9, -8, -7, -6];
+  const maxShift = -20;
+  const expFactor = 0.5;
+  const scrollSpeed = 0.004;
+  const maxDelay = 100; // max scroll delay in px for the first layer
+
+  const endDeg = startDeg.map(
+    (s, i) => +(s + maxShift * Math.pow(expFactor, i)).toFixed(2)
+  );
+
+  const delay = startDeg.map(
+    (_, i) => +(maxDelay * Math.pow(expFactor, i)).toFixed(2)
+  );
+
+  const diffs = startDeg.map((s, i) => Math.abs(endDeg[i] - s));
+  const maxDiff = Math.max(...diffs);
+  const lerp = (a, b, t) => a + (b - a) * t;
+
+  function updateVars() {
+    startDeg.forEach((start, i) => {
+      const effectiveScroll = Math.max(window.scrollY - delay[i], 0);
+      const base = effectiveScroll * scrollSpeed;
+      const t = Math.min(base * (maxDiff / diffs[i]), 1);
+      const rot = lerp(start, endDeg[i], t).toFixed(2);
+      document.body.style.setProperty(`--masthead-rotation-${i + 1}`, `${rot}deg`);
+    });
+  }
+
+  updateVars();
+  window.addEventListener('scroll', updateVars, { passive: true });
+})();
+</script>
+++++
+
+In link:/blog/2025/03/26/design-post/[Part One] we discussed the motivation behind redesigning Jenkins,
+the most used CI/CD platform in the world, and the technical challenges that come with such an effort.
+
+Today, we'll dive deeper into the design decisions behind some of the new UI features, how we're improving the developer
+experience, and what we're doing to ensure compatibility across the plugin ecosystem without slowing down progress.
+
+== Part Two - Lift-off
+
+With Jenkins 2.516.1, we've reworked large amounts of the Jenkins user interface - the result of years of dedicated work
+by the community. This release marks a major milestone in the evolution of Jenkins, with nearly every part of the
+interface having been refined for clarity and usability. From the navigation and theming to pipeline visualization
```

Comment:
> with nearly every part of the interface having been refined for clarity and usability

I'm not a typical Jenkins user but my impression is that this statement is a bit over the top: is only true for Jenkins admins (manage Jenkins area) but not for Jenkins developer users. Developers are mostly interested in the jobs and build pages which did not get a big facelift yet.

Reply:
this is specifically pipeline visualisation - https://github.com/jenkinsci/pipeline-graph-view-plugin/ has had a massive overhaul.

Yes jobs and build pages have more work planned.